File: pc01/pc01_03/python/pc01_03.py
"""
the trip
pc110103
uva10137
"""

import sys
from decimal import Decimal, ROUND_HALF_UP
from typing import List

MAX_STUDENTS = 1000
MAX_MONEY = Decimal('10000.00')

def round_to_cent(value: Decimal) -> Decimal:
    """Round to nearest cent using banker's rounding"""
    return value.quantize(Decimal('0.01'), rounding=ROUND_HALF_UP)

def calculate_exchange(money: List[Decimal], students: int) -> Decimal:
    total = sum(money)
    avg = total / students
    avg_rounded = round_to_cent(avg)
    
    # Amount to be transferred from those who have more
    positive_diff = Decimal('0.00')
    # Amount to be received by those who have less
    negative_diff = Decimal('0.00')
    
    for amount in money:
        diff = round_to_cent(amount - avg_rounded)
        if diff > 0:
            positive_diff += diff
        else:
            negative_diff += -diff
    
    # Cannot generate money, must use minimum amount of transfers
    return min(positive_diff, negative_diff)

def main():
    while True:
        try:
            students = int(input())
            if students == 0:
                break
            
            money = []
            for _ in range(students):
                amount = Decimal(input())
                money.append(round_to_cent(amount))
            
            exchange = calculate_exchange(money, students)
            print(f"${exchange:.2f}")
        
        except (EOFError, ValueError):
            break

if __name__ == '__main__':
    main()

File: pc01/pc01_03/cpp/pc01_03.cpp
/**
 * The trip
 * pc110103
 * uva10137
 */

#include <iostream>
#include <vector>
#include <cmath>
#include <iomanip>
using namespace std;

const int MAX_STUDENTS = 1000;
const double MAX_MONEY = 10000.00;

double roundToCent(double value) {
	return round(value * 100.0) / 100.0;
}

double calculateExchange(vector<double>& money, int students) {
	double total = 0.0;
	for (int i = 0; i < students; i++) {
		total += money[i];
	}

	double avg = total / students;
	double avgRounded = roundToCent(avg);

	double positiveDiff = 0.0;
	double negativeDiff = 0.0;

	for (int i = 0; i < students; i++) {
		double diff = money[i] - avgRounded;
		diff = roundToCent(diff);
		if (diff > 0) {
			positiveDiff += diff;
		} else {
			negativeDiff += -diff;
		}
	}

	return min(positiveDiff, negativeDiff);
}

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);

	cout << fixed << setprecision(2);

	int students;
	vector<double> money(MAX_STUDENTS);

	while (cin >> students && students != 0) {
		for (int i = 0; i < students; i++) {
			cin >> money[i];
			money[i] = roundToCent(money[i]);
		}

		double exchange = calculateExchange(money, students);
		cout << '$' << exchange << '\n';
	}

	return 0;
}